Exiled.Events/Patches/Generic/CameraList.cs

```diff
@@ -0,0 +1,108 @@
+// -----------------------------------------------------------------------
+// <copyright file="CameraList.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+#pragma warning disable SA1402
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using API.Features;
+
+    using Exiled.API.Features.Pools;
+
+    using HarmonyLib;
+
+    using MapGeneration;
+
+    using PlayerRoles.PlayableScps.Scp079;
+    using PlayerRoles.PlayableScps.Scp079.Cameras;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp079InteractableBase.OnRegistered"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(Scp079InteractableBase), nameof(Scp079InteractableBase.OnRegistered))]
+    internal class CameraList
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> codeInstructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(codeInstructions);
+
+            LocalBuilder cameraBase = generator.DeclareLocal(typeof(Scp079Camera));
+            Label ret = generator.DefineLabel();
+
+            // if (this is Scp079Camera camera)
+            //     Room.RoomIdentifierToRoom[Room].CamerasValue.Add(new Camera(camera));
+            newInstructions.InsertRange(
+                newInstructions.Count - 1,
+                new CodeInstruction[]
+                {
+                    new(OpCodes.Ldarg_0),
+                    new(OpCodes.Isinst, typeof(Scp079Camera)),
+                    new(OpCodes.Dup),
+                    new(OpCodes.Stloc_S, cameraBase.LocalIndex),
+                    new(OpCodes.Brfalse_S, ret),
+                    new(OpCodes.Ldsfld, Field(typeof(Room), nameof(Room.RoomIdentifierToRoom))),
+                    new(OpCodes.Ldarg_0),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(Scp079Camera), nameof(Scp079Camera.Room))),
+                    new(OpCodes.Callvirt, Method(typeof(Dictionary<RoomIdentifier, Room>), "get_Item")),
```

Comment:
use PropertyGetter

Reply:
I can be written by `PropertyGetter` instead of `Method`, but it doesn't really matter.

Reply:
it's just to make it more relliable since you will be able to write nameof(Dictionary<RoomIdentifier, Room>.Item)

Reply:
`Item` property is compiler generaded, you can to it like that